modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/ITIgniteNodeRestartTest.java

```diff
@@ -0,0 +1,172 @@
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.runner.app;
+
+import java.util.List;
+import java.util.Map;
+import org.apache.ignite.app.Ignite;
+import org.apache.ignite.app.IgnitionManager;
+import org.apache.ignite.configuration.schemas.network.NetworkConfiguration;
+import org.apache.ignite.internal.app.IgnitionImpl;
+import org.apache.ignite.internal.configuration.ConfigurationManager;
+import org.apache.ignite.internal.manager.IgniteComponent;
+import org.apache.ignite.internal.metastorage.MetaStorageManager;
+import org.apache.ignite.internal.schema.configuration.SchemaConfigurationConverter;
+import org.apache.ignite.internal.testframework.IgniteAbstractTest;
+import org.apache.ignite.schema.ColumnType;
+import org.apache.ignite.schema.SchemaBuilders;
+import org.apache.ignite.schema.SchemaTable;
+import org.apache.ignite.table.Table;
+import org.apache.ignite.table.Tuple;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.commons.util.ReflectionUtils;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+/**
+ * These tests check node restart scenarios.
+ */
+public class ITIgniteNodeRestartTest extends IgniteAbstractTest {
+    /** Test node name. */
+    public static final String NODE_NAME = "TestNode";
+
+    /** Test table name. */
+    public static final String TABLE_NAME = "Table1";
+
+    /**
+     * Restarts empty node.
+     */
+    @Test
+    public void emptyNodeTest() {
+        Ignite ignite = IgnitionManager.start(NODE_NAME, null, workDir.resolve(NODE_NAME));
+
+        IgnitionManager.stop(ignite.name());
+
+        ignite = IgnitionManager.start(NODE_NAME, null, workDir.resolve(NODE_NAME));
+
+        IgnitionManager.stop(ignite.name());
+    }
+
+    /**
+     * Restarts a node with changing configuration.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void changeConfigurationOnStartTest() throws Exception {
+        Ignite ignite = IgnitionManager.start(NODE_NAME, null, workDir.resolve(NODE_NAME));
+
+        int nodePort = getNodePort(NODE_NAME);
+
+        assertEquals(47500, nodePort);
+
+        IgnitionManager.stop(ignite.name());
+
+        String updateCfg =
+            "{\n" +
+            "  \"network\": {\n" +
+            "    \"port\":3322,\n" +
+            "  }\n" +
+            "}";
+
+        ignite = IgnitionManager.start(NODE_NAME, updateCfg, workDir.resolve(NODE_NAME));
+
+        nodePort = getNodePort(NODE_NAME);
+
+        assertEquals(3322, nodePort);
+
+        IgnitionManager.stop(ignite.name());
+    }
+
+    /**
+     * Gets current network port.
+     *
+     * @param nodeName Node name.
+     * @return Network port.
+     * @throws Exception If some issue has happened when getting the port.
+     */
+    private int getNodePort(String nodeName) throws Exception {
```

Comment:
Wow, this is so cursed. Are there other ways of getting this option?

Reply:
I do not know the other way.

Reply:
Reworked with 
```
public class IgniteImpl implements Ignite {
    /**
     * @return Node configuration.
     */
    public ConfigurationRegistry nodeConfiguration() {
        return nodeConfigurationMgr.configurationRegistry();
    }
}
```

Reply:
Thank you!